invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/controls/PublishButton/PublishButton.js

```diff
@@ -54,26 +59,45 @@ class PublishButtonComponent extends Component {
       publishWithoutCommunity,
       formik,
       publishModalExtraContent,
+      permissions,
       ...ui
     } = this.props;
     const { isConfirmModalOpen } = this.state;
     const { values, isSubmitting, handleSubmit } = formik;
 
     const uiProps = _omit(ui, ["dispatch"]);
-
+    const publishDisabled = this.isDisabled(
+      values,
+      isSubmitting,
+      numberOfFiles,
+      permissions
+    );
     return (
       <>
-        <Button
-          disabled={this.isDisabled(values, isSubmitting, numberOfFiles)}
-          name="publish"
-          onClick={this.openConfirmModal}
-          positive
-          icon="upload"
-          loading={isSubmitting && actionState === DRAFT_PUBLISH_STARTED}
-          labelPosition="left"
-          content={buttonLabel}
-          {...uiProps}
-          type="button" // needed so the formik form doesn't handle it as submit button i.e enable HTML validation on required input fields
+        <Popup
+          disabled={!publishDisabled}
+          content={i18next.t(
+            !this.hasPermission(permissions)
+              ? "You don't have permission to publish"
+              : "Required fields are missing"
+          )}
+          wide="very"
```

Comment:
have you checked how this popup behaves in mobile?I am worried for the wide="very"

Reply:
in mobile it would use the normal display method. However, at some point it will switch to `very' and then overlaps with the left/middle column of the deposit page. Removed the wide attribute.